third_party/3/pandas/core/frame.pyi

```diff
@@ -184,8 +184,15 @@ class DataFrame(NDFrame):
     def unstack(self, level: Union[Level, List[Level]] = ..., fill_value: Optional[Scalar] = ...) -> FrameOrSeries: ...
     def melt(self, id_vars: Optional[Union[Tuple[Column], List[Column], np.ndarray]] = ..., value_vars: Optional[Union[Sequence[Column], np.ndarray]] = ..., var_name: Optional[Scalar] = ..., value_name: Scalar = ..., col_level: Optional[Level] = ..., ignore_index: bool = ...) -> DataFrame: ...
     def diff(self, periods: int = ..., axis: AxisOption = ...) -> DataFrame: ...
+    # Less precise type to nudge folks towards `.agg`
     def aggregate(self, func: AggregationFunction = ..., axis: AxisOption = ..., *args: Any, **kwargs: Any) -> Union[Scalar, FrameOrSeries]: ...
-    def agg(self, func: AggregationFunction = ..., axis: AxisOption = ..., *args: Any, **kwargs: Any) -> Union[Scalar, FrameOrSeries]: ...
+    @overload
+    def agg(self, func: MultipleAggregationFunctions = ..., axis: AxisOption = ..., *args: Any, **kwargs: Any) -> DataFrame: ...
+    @overload
+    def agg(self, func: Union[Callable[..., Any], str], axis: AxisOption = ..., *args: Any, **kwargs: Any) -> Series: ...
+    # If numpy is not available to mypy, np.func will be `Any` and can't be used for overloading
+    @overload
```

Comment:
Instead of overloading, could we have a one extended version of AggregationFunctions?

Reply:
Not really because the return types differ.